op-supervisor/supervisor/backend/syncnode/reset.go

```diff
@@ -0,0 +1,317 @@
+package syncnode
+
+import (
+	"context"
+	"errors"
+
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/types"
+)
+
+// ongoingReset represents a bisection,
+// where a is the earliest in the range,
+// and z is the latest in the range.
+// when a and z are directly adjacent, the
+// bisection is complete. a is the last consistent
+// block, and z is the first inconsistent block.
+type ongoingReset struct {
+	a eth.BlockID
+	z eth.BlockID
+}
+
+func (m *ManagedNode) resetIfAhead() {
```

Comment:
It would be great if we can encapsulate the bisection methods and relevant sync-node state into its own object

Reply:
Sure, I've encapsulated most of the functionality to a struct called `resetTracker`. The `resetIf...` functions and the last seen heads are still handled by the Managed Node, but once an inconsistency or reason to reset is determined, `m.resetTracker.beginBisectionReset(z)` kicks off the process of identifying the range (using z as the end-of-range), bisecting, determining heads, and then finally calls back to the managed node to reset.